Problem2/src/problem2/Problem2.java

```diff
@@ -1,16 +1,54 @@
 package problem2;
 
-/**
- *
- * @author cgallinaro
- */
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.*;
+
 public class Problem2 {
 
-    /**
-     * @param args the command line arguments
-     */
     public static void main(String[] args) {
-        // TODO code application logic here
+        Scanner input = new Scanner(System.in);
+        StringBuilder s1 = new StringBuilder();
+        System.out.println("Enter a number");
+        int size = input.nextInt();
+        boolean[] primes = new boolean[size + 1];
+        Arrays.fill(primes, true);
+        int divider = 2;
+        int result;
+        for (int i = 2; i < primes.length; i++) {
+            int add = 0;
+            if (primes[divider] == true) {
+                for (int e = 2; e < primes.length; e++) {
```

Comment:
You get the right answer, but your algorithm is not quite right.   If the number is prime, you’re supposed to count by multiples of the prime number to update the array.